### DISCUSSION TITLE
Using firebase-auth with custom UI?

### DISCUSSION THREAD

**Discussion by dflorey** [firebase/firebase-js-sdk#4504] [Q&A] [answered]
Is there a way to use the new alpha auth SDK with a custom UI (popup flow) or do we have to wait for the new firebase-ui alpha?
It would be great to get rid of the old firebase-ui deps as the bloat up the bundle size, but still get some functionality to simplify the popup flow...

**Comment by sam-gc** [comment, marked as answer]
Hi @dflorey, thanks for reaching out! What features would you specifically like to see / what complexity would you like to have reduced? The modular SDK popup flow follows the same logic / structure as the old SDK.

**Reply by dflorey** [reply]
Probably I'm just confused :-/
As far as I remember the old firebase-ui comes with a lot of stuff that bloats up the bundle size (material light components, polyfills etc.)
I'd like to be able to use my own buttons instead (e.g. Login with github) while leveraging the modular firebase-auth package.
I did not find any example in the docs how to do that. When using the popup flow (SPA) I guess there is some additional logic involved (checking if popup window is still open etc.) that is performed by firebase-ui.
It would be nice if there would be an example how to build a minimal app with custom login buttons using firebase auth / popup flow.

**Reply by sam-gc** [reply]
No worries! Firebase UI does a lot of heavy lifting for certain tasks (automatic account linking, for example) as well as helping with the sign in buttons and all that (this is especially helpful for other flows like email/password or phone, which can be a bit of work to set up yourself).

[Here is an example for GitHub w/ Popup](https://modularfirebase.web.app/common-use-cases/authentication/#sideNav:~:text=GitHub%20sign%20in%20with%20popup) (I see the heading for the whole section is wrong, we will fix that :)

**Reply by akauppi** [reply]
@dflorey I got the `aside-keys` package converted to @exp API today. It works, has plain HTML demo, is shipped as a web component, and has no need for Firebase UI. Maybe it serves as a platform for you?

https://github.com/akauppi/aside-keys
I seem to have forgotten to push the @exp branch to GitHub - will fix that in 10hrs.

Initially, only Google social sign-in but I’d like to have more, based on demand.

**Reply by akauppi** [reply]
The package is now updated.

**Reply by dflorey** [reply]
Thanks! 
I'm still not sure about the features provided by the "pure" auth package vs. firebase-ui.
It looks as if I can just invoke signInWithPopup from my own button without needing the firebase-ui package at all.
Very weird. Would be great if the docs could clarify why/when to use firebase-ui.
Makes my app much smaller and more consistent :-)

**Reply by sam-gc** [reply]
@dflorey That's a fair point, sorry for the confusion. The high-level bullet points on [the Firebase UI docs](https://firebase.google.com/docs/auth/web/firebaseui) are the main features you get by using Firebase UI (namely: things like localization, linking, easy UI buttons for multiple providers, etc).

But yeah with all of that said, there are some UI aspects in the core SDK (namely the popup flow, recaptcha, etc).

**Reply by akauppi** [reply]
@samhorlbeck I second @dflorey on this 100%. I too was very surprised to see `signInWithPopup` just work. Was expecting that I need to crawl through FirebaseUI sources to duplicate some functionality. But that wasn't needed.

I think the best place to clarify this would be in the Firebase Auth docs; not FirebaseUI.  At the same time, the few references to FirebaseUI in official Firebase documentation could be reviewed.

**Reply by sam-gc** [reply]
:+1: I will chat with the team
